package com.cartethyia.easyorange.user.adapter.inbound.web.dto.response;

import com.cartethyia.easyorange.common.constant.CommonConstant;
import com.cartethyia.easyorange.user.domain.enums.UserType;
import com.fasterxml.jackson.annotation.JsonFormat;
import java.time.LocalDateTime;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class UserProfileResponse implements CommonUserFields {

    private String userId;

    private String username;

    private String nickname;

    private String email;

    private String phone;

    private String realName;

    private String status;

    private String statusDesc;

    private String gender;

    private UserType userType;

    private String avatar;

    @JsonFormat(pattern = CommonConstant.DATETIME_FORMAT, timezone = "GMT+8")
    private LocalDateTime createTime;

    @JsonFormat(pattern = CommonConstant.DATETIME_FORMAT, timezone = "GMT+8")
    private LocalDateTime updateTime;
}
